src/resources/views/fields/address_google.blade.php

```diff
@@ -0,0 +1,119 @@
+                        @include('crud::inc.field_attributes')
+                >
+            @endif
+            @if(isset($field['suffix']))
+                <div class="input-group-addon">{!! $field['suffix'] !!}</div> @endif
+            @if(isset($field['prefix']) || isset($field['suffix'])) </div> @endif
+
+    {{-- HINT --}}
+    @if (isset($field['hint']))
+        <p class="help-block">{!! $field['hint'] !!}</p>
+    @endif
+</div>
+
+{{-- Note: you can use  to only load some CSS/JS once, even though there are multiple instances of it --}}
+
+{{-- ########################################## --}}
+{{-- Extra CSS and JS for this particular field --}}
+{{-- If a field type is shown multiple times on a form, the CSS and JS will only be loaded once --}}
+@if ($crud->checkIfFieldIsFirstOfItsType($field, $fields))
+
+    {{-- FIELD CSS - will be loaded in the after_styles section --}}
+    @push('crud_fields_styles')
+        <style>
+            .ap-input-icon.ap-icon-pin {
+                right: 5px !important;
+            }
+
+            .ap-input-icon.ap-icon-clear {
+                right: 10px !important;
+            }
+        </style>
+    @endpush
+
+    {{-- FIELD JS - will be loaded in the after_scripts section --}}
+    @push('crud_fields_scripts')
+        <script>
+
+            //Function that will be called by Google Places Library
+            function initAutocomplete() {
+
+
+                $('[data-address]').each(function () {
+
+                    var $this = $(this),
+                        $addressConfig = $this.data('address'),
+                        $field = $('[name="' + $addressConfig.field + '"]');
+
+                    if ($field.val().length) {
+                        var existingData = JSON.parse($field.val());
+                        $this.val(existingData.value);
+                    }
+
+                    var $autocomplete = new google.maps.places.Autocomplete(
+                        ($this[0]),
+                        {types: ['geocode']});
+
+                    $autocomplete.addListener('place_changed', function fillInAddress() {
+
+                        var place = $autocomplete.getPlace();
+                        var value = $this.val();
+                        var latlng = place.geometry.location;
+                        var data = {"value": value, "latlng": latlng};
+
+                        for (var i = 0; i < place.address_components.length; i++) {
+                            var addressType = place.address_components[i].types[0];
+                                data[addressType] = place.address_components[i]['long_name'];
+                        }
+                        $field.val(JSON.stringify(data));
+
+                    });
+
+
+                });
+
+            }
+
+        </script>
+        <script src="https://maps.googleapis.com/maps/api/js?key={{env('GOOGLE_PLACES_API_KEY')}}&libraries=places&callback=initAutocomplete"
```

Comment:
`env` vs `config`? Not sure.

Reply:
You should not use env() ouside of config files. 

Please add a key on a config file that pull's the value from .env file and use config() helper instead to retrieve the value.

Thanks,
Pxpm

Reply:
@alanbrande as pxpm said - you should *NEVER* use `env()` outside of your config files, this is because when you run `artisan config:cache` laravel then returns NOTHING from the `env()` function - this means your code would return `i/js?key=NULL&libr` which would not be the desired affect.

You should make a new item within a config, suggest maybe `config/services.php` maybe if there isn't a suitable place in backpack - this config should pull from `env()` just like the database stuff does. This means that when you run `config:cache` laravel will read your env file, cache the result making it accessible later on.

Reply:
Done!

Reply:
Thank you @lloy0076 , @OwenMelbz and @pxpm for the feedback - I think you’re all right :-)